vllm/config.py

```diff
@@ -4346,6 +4346,41 @@ def set_splitting_ops_for_v1(self):
             ]
 
 
+class ReasoningConfig:
+    """Configuration for reasoning models."""
+
+    think_start_str: Optional[str] = None
+    """String that indicates the start of reasoning."""
+    think_end_str: Optional[str] = None
+    """String that indicates the end of reasoning."""
+    think_start_token_ids: Optional[int] = None
+    """Token ID that indicates the start of reasoning."""
+    think_end_token_ids: Optional[int] = None
+    """Token ID that indicates the end of reasoning."""
+    low_effort_token_budget: Optional[int] = 1024
+    """Token budget for low effort reasoning."""
+    medium_effort_token_budget: Optional[int] = 2048
+    """Token budget for medium effort reasoning."""
+    high_effort_token_budget: Optional[int] = 8192
+    """Token budget for high effort reasoning."""
+
+    def __init__(self,
+                 think_start_str: Optional[str] = None,
+                 think_end_str: Optional[str] = None,
+                 think_start_token_ids: Optional[int] = None,
+                 think_end_token_ids: Optional[int] = None,
+                 low_effort_token_budget: Optional[int] = 1024,
+                 medium_effort_token_budget: Optional[int] = 2048,
+                 high_effort_token_budget: Optional[int] = 8192):
+        self.think_start_str = think_start_str
+        self.think_end_str = think_end_str
+        self.think_start_token_ids = think_start_token_ids
+        self.think_end_token_ids = think_end_token_ids
+        self.low_effort_token_budget = low_effort_token_budget
+        self.medium_effort_token_budget = medium_effort_token_budget
+        self.high_effort_token_budget = high_effort_token_budget
```

Comment:
If we're going to add a new config please:

- make it a `dataclass` like the rest of the configs
- make it a `config` like the rest of the configs
- Don't type hint `Optional` if the default is not `None`

```suggestion
@config
@dataclass
class ReasoningConfig:
    """Configuration for reasoning models."""

    think_start_str: Optional[str] = None
    """String that indicates the start of reasoning."""
    think_end_str: Optional[str] = None
    """String that indicates the end of reasoning."""
    think_start_token_ids: Optional[int] = None
    """Token ID that indicates the start of reasoning."""
    think_end_token_ids: Optional[int] = None
    """Token ID that indicates the end of reasoning."""
    low_effort_token_budget: int = 1024
    """Token budget for low effort reasoning."""
    medium_effort_token_budget: int = 2048
    """Token budget for medium effort reasoning."""
    high_effort_token_budget: int = 8192
    """Token budget for high effort reasoning."""
```

Reply:
Thank you for your comments. I didn't know much about those best practices.
I have made all the updates.